src/api/types/types.ts

```diff
@@ -1,100 +1,134 @@
-/* eslint-disable @typescript-eslint/no-explicit-any */
-import { Context } from "@deskpro/app-sdk";
-import { Reducer } from "react";
 import { ADFEntity } from "@atlaskit/adf-utils";
-import { IssueMeta } from "../../types";
-
-export type ApiRequestMethod = "GET" | "POST" | "PUT" | "DELETE";
-
-export type StoreReducer = Reducer<State, Action>;
-
-export type Dispatch = (action: Action) => void;
-
-export type Page =
-  | "home"
-  | "link"
-  | "view"
-  | "create"
-  | "edit"
-  | "comment"
-  | "view_permissions"
-  | "verify_settings";
-
-export interface State {
-  page?: Page;
-  pageParams?: any;
-  context?: TicketContext;
-  linkIssueSearchResults?: { loading: boolean; list: IssueSearchItem[] };
-  linkedIssuesResults?: { loading: boolean; list: IssueItem[] };
-  linkedIssueAttachments?: {
-    loading: boolean;
-    list: { [key: string]: IssueAttachment[] };
+import { components } from "./schema";
+import { RequiredProps, DateTime, FieldType } from "../../types";
+import { Issuetype, Project, Watches, Priority, UserBean, Components, Progress, Status, Votes } from "./fieldsValue";
+import { SprintCustomValue, CustomFieldsValues, CustomFieldValue } from "./customFieldsValue";
+
+export type Schemas = components["schemas"];
+
+export type IssuePicker = Required<components["schemas"]["SuggestedIssue"]>;
+
+export type PickerSection = Omit<
+  components["schemas"]["IssuePickerSuggestionsIssueType"],
+  "id"|"issues"
+> & {
+  id: string;
+  issues: IssuePicker[];
+};
+
+export type IssuesPicker = Omit<components["schemas"]["IssuePickerSuggestions"], "sections"> & {
+  sections: PickerSection[];
+};
+
+export type FieldsValues = {
+  description: ADFEntity;
+  summary: string;
+  issuetype: Issuetype;
+  project: Project;
+  workratio: number;
+  watches: Watches;
+  priority: Priority;
+  labels: string[];
+  fixVersions: [{ id: number }];
```

Comment:
This looks incorrect, it should probably be `fixVersions: {id: number}[]` as there might not be a fix version (zero records) or could be part of many (multiple records) - eitherway, it's not exactly always one.

Reply:
fixed